### PR TITLE
lua usermanual : split API into its own document

### PR DESCRIPTION
**Pull request by boucman**
@hanatos this commit should create a new document similar to usermanual, I have tested that it generates correctly html, pdf and PHP for wordpress, but I can't test if/how it integrates in the website

could you have a look at this and if all is good, merge the PR ? thx

### CODE REVIEW

**Comment by upegelow** [comment]
Having a first look I discovered an issue with some dangling cross references in lua.xml. These cross references should be replaced by an <ulink> to the manual. We don't have the specific place yet, but it will certainly end up under http://www.darktable.org/resources/

**Comment by boucman** [comment]
good point

File: doc/usermanual/CMakeLists.txt [hunk 1]
```diff
@@ -9,8 +9,10 @@ message(STATUS ${IMAGES_PATHS})
 
 if(BUILD_USERMANUAL)
 	add_custom_target(usermanual ALL)
+	add_custom_target(api ALL)
```
Comment (houz) [inline doc/usermanual/CMakeLists.txt]: Maybe call it something else, these target names are going to the global namespace, and something like `api` could be needed for other things. And it's not that obvious what it does. How about `apimanual` instead?

File: doc/usermanual/CMakeLists.txt [hunk 2]
```diff
@@ -281,9 +283,137 @@ if(can_build_usermanual)
...
 	endif()
+#
+# LUA API GENERATION
+#
+
+	#
+	# draft pdf API
+	#
+	add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/darktable-api.pdf
```
Comment (houz) [inline doc/usermanual/CMakeLists.txt]: `darktable-lua-api.pdf` might be better, once day we might add a C API, or finally document the internal one ...

File: doc/usermanual/CMakeLists.txt [hunk 3]
```diff
@@ -281,9 +283,137 @@ if(can_build_usermanual)
...
+		COMMAND xsltproc --output ${CMAKE_CURRENT_BINARY_DIR}/darktable-api.fo xsl/darktable_fo.xsl  ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml
+		COMMAND fop -c fopconfig.xml ${CMAKE_CURRENT_BINARY_DIR}/darktable-api.fo -pdf ${CMAKE_CURRENT_BINARY_DIR}/darktable-api.pdf
+		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+		DEPENDS  ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml
+		COMMENT "Building API" VERBATIM
```
Comment (houz) [inline doc/usermanual/CMakeLists.txt]: Should also be rephrased.

File: doc/usermanual/CMakeLists.txt [hunk 4]
```diff
@@ -281,9 +283,137 @@ if(can_build_usermanual)
...
+	)
+	add_custom_target(darktable-api DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/darktable-api.pdf)
+	add_dependencies(darktable-api target_media_images )
```
Comment (houz) [inline doc/usermanual/CMakeLists.txt]: Does your API doc need the images of the usermanual?

File: doc/usermanual/CMakeLists.txt (hunk 1 above)
Comment (houz) [inline doc/usermanual/CMakeLists.txt]: Oh, and you only add the translated docs to the `api` target, I guess the regular ones should be built there, too?

Comment (boucman) [inline doc/usermanual/CMakeLists.txt]: agreed, renaming to luaapi, 

however, only translated UM are added to the UM target and I did the same... I can change that, but i'd rather stay consistant with UM

File: doc/usermanual/CMakeLists.txt (hunk 3 above)
Comment (boucman) [inline doc/usermanual/CMakeLists.txt]: not sure what you mean, changed from darktable-api to darktable-lua-api ?

File: doc/usermanual/CMakeLists.txt (hunk 4 above)
Comment (boucman) [inline doc/usermanual/CMakeLists.txt]: no, but most people will build both the api and the UM, so no harm... and i might add images at some point, so I might as well have the infra in place

File: doc/usermanual/CMakeLists.txt [hunk 5]
```diff
@@ -286,9 +288,137 @@ if(can_build_usermanual)
...
+	#
+	# draft pdf API
+	#
+	add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api.pdf
+		COMMAND xsltproc --output ${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api.fo xsl/darktable_fo.xsl  ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml
+		COMMAND fop -c fopconfig.xml ${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api.fo -pdf ${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api.pdf
+		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+		DEPENDS  ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml
+		COMMENT "Building API" VERBATIM
```
Comment (LebedevRI) [inline doc/usermanual/CMakeLists.txt]: Maybe
```
COMMENT "Building Lua API" VERBATIM
```
?

File: doc/usermanual/CMakeLists.txt [hunk 6]
```diff
@@ -286,9 +288,137 @@ if(can_build_usermanual)
...
+        if (Saxon_FOUND)
+	    add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/html_api/index.html
+                COMMAND ${XSLT_SAXON_COMMAND} ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml xsl/darktable_html_single_ext.xsl chunker.output.encoding=UTF-8 base.dir=${CMAKE_CURRENT_BINARY_DIR}/html_api/
+		COMMAND find ${CMAKE_CURRENT_BINARY_DIR}/html_api -name *.html -exec sed -n -i "1h;1!H;\${;g;s/${binary_path}\\/media\\///g;p;}" {} \;
+		COMMAND find ${CMAKE_CURRENT_BINARY_DIR}/html_api -name *.html -exec sed -n -i "1h;1!H;\${;g;s/<body[^>]*>/<body class='usermanual' onLoad='parent.onContentLoad(this);'>/g;p;}" {} \;
+		COMMAND cp -r ${CMAKE_CURRENT_BINARY_DIR}/media/images ${CMAKE_CURRENT_BINARY_DIR}/html_api/
+		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+		DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml ${CMAKE_CURRENT_BINARY_DIR}/html_api/usermanual.css
+		COMMENT "Building html API" VERBATIM
```
Comment (LebedevRI) [inline doc/usermanual/CMakeLists.txt]: Same

File: doc/usermanual/CMakeLists.txt [hunk 7]
```diff
@@ -286,9 +288,137 @@ if(can_build_usermanual)
...
+        else (Saxon_FOUND)
+            add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/html_api/index.html
+                COMMAND ${XSLT_XSLTPROC_EXECUTABLE} --stringparam chunker.output.encoding UTF-8 --stringparam base.dir ${CMAKE_CURRENT_BINARY_DIR}/html_api/ xsl/darktable_html_single.xsl ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml
+		COMMAND find ${CMAKE_CURRENT_BINARY_DIR}/html_api -name *.html -exec sed -i s/${binary_path}\\/media\\///g {} \;
+		COMMAND find ${CMAKE_CURRENT_BINARY_DIR}/html_api -name *.html -exec sed -i "s/<body[^>]*>/<body class='usermanual' onLoad='parent.onContentLoad(this);'>/g" {} \;
+		COMMAND cp -r ${CMAKE_CURRENT_BINARY_DIR}/media/images ${CMAKE_CURRENT_BINARY_DIR}/html_api/
+		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+		DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml ${CMAKE_CURRENT_BINARY_DIR}/html_api/usermanual.css
+		COMMENT "Building html API" VERBATIM
```
Comment (LebedevRI) [inline doc/usermanual/CMakeLists.txt]: Same

File: doc/usermanual/CMakeLists.txt [hunk 8]
```diff
@@ -286,9 +288,137 @@ if(can_build_usermanual)
...
+	add_dependencies(darktable-lua-api-html target_media_images)
+	install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html_api DESTINATION ./share/doc/darktable/luaapi OPTIONAL)
+
+	#
+	# draft chunked usermanual for wordpress
+	#
 
+        if (Saxon_FOUND)
+            add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/wp_api/index.html.php
```
Comment (LebedevRI) [inline doc/usermanual/CMakeLists.txt]: Shouldn't it be under e.g. wp_lua_api ?

File: doc/usermanual/CMakeLists.txt [hunk 9]
```diff
@@ -286,9 +288,137 @@ if(can_build_usermanual)
...
+	#
+	if(NOT ${Xml2po_BIN} STREQUAL "Xml2po_BIN-NOTFOUND")
+		# localized draft usermanuals
+		file(STRINGS "po/LINGUAS" LANGUAGES)
+		foreach(language ${LANGUAGES})
+			set(pofile "po/${language}.po")
+			set(pdffile "${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api-${language}.pdf")
+			set(fofile "${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api-${language}.fo")
+			set(xmlfile "${CMAKE_CURRENT_SOURCE_DIR}/lua/lua_api.xml")
+			set(xmllangfile "${CMAKE_CURRENT_BINARY_DIR}/darktable_api-${language}.xml")
```
Comment (LebedevRI) [inline doc/usermanual/CMakeLists.txt]:
```
set(xmllangfile "${CMAKE_CURRENT_BINARY_DIR}/darktable-lua-api-${language}.xml")
```

**Comment by boucman** [comment]
Ok, except for crosslinks to the API everything should be ready(i'm not sure how to do that @upegelow if you have any idea, just tell me in a comment and I'll update everything. There seems to be only a couple of links in doc/usermanual/lua/lua.xml)

**Comment by boucman** [comment]
commited directly to trunk

**Pull request closed without merging**